ironfish/src/blockchain/blockchain.test.perf.ts

```diff
@@ -61,104 +57,195 @@ describe('Blockchain', () => {
-        // the blocks from blocksB to be removed from the trees
-        const startRewind = Date.now()
-        await node.chain.addBlock(blocksB[forkLength - 1])
-        const endRewind = Date.now()
-        samplesRewind.push(endRewind - startRewind)
-
-        const endAll = Date.now()
-        samplesAll.push(endAll - startAll)
-
-        // Verify the head is the last block in blocksB
-        const actualHead = node.chain.head
-        const expectedHead = blocksB[forkLength - 1]
-        Assert.isNotNull(actualHead, 'Chain has no head')
-        expect(actualHead.hash.toString('hex')).toEqual(
-          expectedHead.header.hash.toString('hex'),
-        )
+    Assert.isEqual(balanceA.confirmed, BigInt(1999999901))
+    Assert.isEqual(balanceA.unconfirmed, BigInt(1999999901))
+    Assert.isEqual(balanceB.confirmed, BigInt(1999999901))
+    Assert.isEqual(balanceB.unconfirmed, BigInt(1999999901))
+  })
+
+  it('Times Ran: 5, Fork Length: 1', async () => {
+    const result = await runTest(5, 1)
+    printResults(result)
+
+    const totalTestAverageTime = MathUtils.arrayAverage(result.all)
+    expect(totalTestAverageTime).toBeLessThanOrEqual(39.4)
+
+    const addBlockLinearTime = MathUtils.arrayAverage(result.add)
+    expect(addBlockLinearTime).toBeLessThanOrEqual(1)
+
+    const addBlockOnForkTime = MathUtils.arrayAverage(result.fork)
+    expect(addBlockOnForkTime).toBeLessThanOrEqual(1)
+
+    const addHeadRewindForkBlocksTime = MathUtils.arrayAverage(result.rewind)
+    expect(addHeadRewindForkBlocksTime).toBeLessThanOrEqual(36.6)
+  })
+
+  it('Times Ran: 5, Fork Length: 3', async () => {
+    const result = await runTest(5, 3)
+    printResults(result)
+
+    const totalTestAverageTime = MathUtils.arrayAverage(result.all)
+    expect(totalTestAverageTime).toBeLessThanOrEqual(274.6)
+
+    const addBlockLinearTime = MathUtils.arrayAverage(result.add)
+    expect(addBlockLinearTime).toBeLessThanOrEqual(45.3)
+
+    const addBlockOnForkTime = MathUtils.arrayAverage(result.fork)
+    expect(addBlockOnForkTime).toBeLessThanOrEqual(34.2)
+
+    const addHeadRewindForkBlocksTime = MathUtils.arrayAverage(result.rewind)
+    expect(addHeadRewindForkBlocksTime).toBeLessThanOrEqual(157.8)
+  })
+
+  it('Times Ran: 5, Fork Length: 5', async () => {
+    const result = await runTest(5, 5)
+    printResults(result)
+
+    const totalTestAverageTime = MathUtils.arrayAverage(result.all)
+    expect(totalTestAverageTime).toBeLessThanOrEqual(612.2)
+
+    const addBlockLinearTime = MathUtils.arrayAverage(result.add)
+    expect(addBlockLinearTime).toBeLessThanOrEqual(57.3)
+
+    const addBlockOnForkTime = MathUtils.arrayAverage(result.fork)
+    expect(addBlockOnForkTime).toBeLessThanOrEqual(75.1)
+
+    const addHeadRewindForkBlocksTime = MathUtils.arrayAverage(result.rewind)
+    expect(addHeadRewindForkBlocksTime).toBeLessThanOrEqual(82.6)
+  })
+
+  it('Times Ran: 5, Fork Length: 10', async () => {
+    const result = await runTest(5, 10)
```

Comment:
Are we planning on using the timing of `it(` blocks as the final result of performance tests? If so I think we should minimize code that's called in this block besides the `runTest` function so we get accurate numbers

Reply:
Moved the node set up outside it()